modules/configuration/src/main/java/org/apache/ignite/internal/configuration/ConfigurationChanger.java

```diff
@@ -251,30 +225,27 @@ public void initialize(ConfigurationType storageType) throws ConfigurationValida
                 throw (ConfigurationChangeException)cause;
 
             throw new ConfigurationChangeException(
-                "Failed to write defalut configuration values into the storage " + configurationStorage.getClass(), e
+                "Failed to write defalut configuration values into the storage " + storage.getClass(), e
             );
         }
         catch (InterruptedException e) {
             throw new ConfigurationChangeException(
-                "Failed to initialize configuration storage " + configurationStorage.getClass(), e
+                "Failed to initialize configuration storage " + storage.getClass(), e
             );
         }
     }
 
     /**
      * Changes the configuration.
-     * @param source Configuration source to create patch from.
-     * @param storage Expected storage for the changes. If null, a derived storage will be used
-     * unconditionaly.
+     *
+     * @param src Configuration source to create patch from.
      * @return Future that is completed on change completion.
      */
-    public CompletableFuture<Void> change(
-        ConfigurationSource source,
-        @Nullable ConfigurationStorage storage
-    ) {
+    public CompletableFuture<Void> change(ConfigurationSource src) {
         Set<ConfigurationType> storagesTypes = new HashSet<>();
 
         ConstructableTreeNode collector = new ConstructableTreeNode() {
```

Comment:
I think that this all monstrosity with the collector should be removed or heavily cut. We don't need this logic anymore since changer only knows about one storage